### PR TITLE
FOGL-5402 disallow default shippped certs deletion and non-admin should not be allowed to overwrite the configured ca/https and default admin certs

### PR DESCRIPTION
**Pull request by praveen-garg**

### CODE REVIEW

**Comment by praveen-garg** [comment]
@ashish-jabble can you take care of unit tests and also see if anything I missed?

File: python/fledge/services/core/api/certificate_store.py [hunk 1]
```diff
@@ -158,16 +184,20 @@ async def delete_certificate(request):
...
+            _logger.warning(FORBIDDEN_MSG)
+            raise web.HTTPForbidden(reason=FORBIDDEN_MSG, body=json.dumps({"message": FORBIDDEN_MSG}))
+    
     cf_mgr = ConfigurationManager(connect.get_storage_async())
-    result = await cf_mgr.get_category_item(category_name='rest_api', item_name='certificateName')
-    if file_name.split('.')[0] == result['value']:
-        msg = 'Certificate with name {} is already in use, you can not delete'.format(file_name)
+    cat = await cf_mgr.get_category_all_items(category_name='rest_api')
+    _logger.warning("---")
+    _logger.warning(cat)
```
Comment (praveen-garg) [inline python/fledge/services/core/api/certificate_store.py:196]: remove those extra log statements too.

File: tests/unit/python/fledge/services/core/api/test_certificate_store.py [hunk 2]
```diff
@@ -244,7 +244,6 @@ async def test_bad_delete_cert_with_invalid_filename(self, client, cert_name, ac
     
     @pytest.mark.parametrize("cert_name, actual_code, actual_reason", [
         ('root.txt', 400, "Accepted file extensions are ('.cert', '.cer', '.crt', '.json', '.key', '.pem')"),
-        ('admin.key', 400, "Admin certs cannot be deleted"),
```
Comment (praveen-garg) [inline tests/unit/python/fledge/services/core/api/test_certificate_store.py:247]: key canbe deleted.

admin once copied/ pulled from foglamp can also delete the admin (.cert) certs from foglamp, but disallow to normal user and when auth is optional.

**Comment by ashish-jabble** [comment]
> @ashish-jabble can you take care of unit tests and also see if anything I missed?

Existing tests fixed and also added new test case scenarios as per new changes; in PR https://github.com/fledge-iot/fledge/pull/395

File: python/fledge/services/core/api/certificate_store.py [hunk 3]
```diff
@@ -79,17 +82,41 @@ async def upload(request):
...
+        cat = await cf_mgr.get_category_all_items(category_name='rest_api')
+        configured_ca_and_tls_certs = [cat['certificateName']['value'], cat['authCertificateName']['value']]
+        if cert_filename and cert_filename.rpartition('.')[0] in configured_ca_and_tls_certs:  # we better disallow any extension with those names instead of [1]/endswith .cert
+            msg = 'Certificate with name {} is configured to be used, ' \
+                  'An `admin` role permissions required to add/overwrite.'.format(cert_filename)
+            _logger.warning(msg)
+            raise web.HTTPForbidden(reason=msg, body=json.dumps({"message": msg}))
+
     key_valid_extensions = ('.key', '.pem')
     cert_valid_extensions = ('.cert', '.cer', '.crt', '.json', '.pem')
```
Comment (praveen-garg) [inline python/fledge/services/core/api/certificate_store.py:121]: @ashish-jabble Please file a jira to  allow .p12, .pfx and .der extensions. May be .csr/.crl too.

**Review by MarkRiddoch — APPROVED** [review]

File: python/fledge/services/core/api/certificate_store.py (hunk 3 above)
Comment (ashish-jabble) [reply python/fledge/services/core/api/certificate_store.py:121]: FOGL-5480 JIRA raised.

**Pull request merged**